pkg/cortex/serve/cortex_internal/lib/api/batching.py

```diff
@@ -32,16 +33,14 @@ def __init__(self, predictor_impl: Callable, max_batch_size: int, batch_interval
         self.batch_max_size = max_batch_size
         self.batch_interval = batch_interval  # measured in seconds
 
-        # waiter prevents new threads from modifying the input batch while a batch prediction is in progress
-        self.waiter = td.Event()
-        self.waiter.set()
-
-        self.barrier = td.Barrier(self.batch_max_size + 1, action=self.waiter.clear)
+        self.barrier = td.Barrier(self.batch_max_size + 1)
 
         self.samples = {}
         self.predictions = {}
         td.Thread(target=self._batch_engine).start()
 
+        self.thread_id_generator = itertools.count()
```

Comment:
doesn't the thread already have a unique ID? my concern is that this number grows forever and it might be a problem after a while

Reply:
You're right

I wanted to put aside all the possible causes but I now think that `td.get_ident` was OK to generate a thread ID.

`itertools.count` used this way actually generates a unique ID per sample, so if kept I think that `thread_id*`should be renamed to `sample_id*`

I think that the infinite growth of the IDs generated by `itertools.count` is not a problem in Python 3 (even with Python 2, the service would work for decades with billions of requests per second)

Reply:
Yeah, I'll rename that to `sample_id*`.